SuiteCRM/SuiteCRM
# Feature Anti Malware Scanner Integration

## Description



After experimenting with clamav and sophos anti-virus for linux, I came up with a framework to integrate anti-malware software with suitecrm.

Using the PSR-4 autoloader you can configure first-party, custom, and third party packages. SuiteCRM can then autoload those packages and run a command line tool to scan uploads for malware.

![av-config](https://user-images.githubusercontent.com/12231216/36887872-f94c6364-1dea-11e8-9907-40fc57113c29.png)
Config.php

The configure can be used to customise the executable path and the option passed into the command line tool.

Developers can create providers which implement the AntiMalwareFileScanner interface and throw a MalwareFound exception. This will make it really easy to support many command line based antivirus programs.

Incidents are recorded in the SuiteCRM.log
![av-log](https://user-images.githubusercontent.com/12231216/36887999-85a43468-1deb-11e8-97aa-50e2c511f9c2.png)


## Motivation and Context

- Scans uploaded content with anti malware software and blocks any issues found
- Adds improved exception handling in the sugar controller, so that better error messages are reported back to the user, if a suitecrm exception is thrown in the GUI.


## How To Test This

- Install [clamav](https://www.clamav.net/) or [sophos](https://www.sophos.com/en-us/products/free-tools/sophos-antivirus-for-linux.aspx) on the web server
- Rebuild the config
- Configure the path location
- Set enabled to true
- Upload a file, create a document/note with the [EICAR](http://www.eicar.org/86-0-Intended-use.html) test file

(Optional)
- Run the automated unit test suite see tests/unit/lib/AntiMalware

## Types of changes


### Final checklist
